processor/transformer/transformer.go

```diff
@@ -395,27 +393,22 @@ func (trans *handle) request(ctx context.Context, url, stage string, data []Tran
 	}
 
 	var transformerResponses []TransformerResponse
-	if statusCode == http.StatusOK {
+	switch statusCode {
+	case http.StatusOK:
 		integrations.CollectIntgTransformErrorStats(respData)
 
 		trace.Logf(ctx, "Unmarshal", "response raw size: %d", len(respData))
 		trace.WithRegion(ctx, "Unmarshal", func() {
 			err = json.Unmarshal(respData, &transformerResponses)
 		})
 		// This is returned by our JS engine so should  be parsable
-		// but still handling it
+		// Panic the processor to avoid replays
 		if err != nil {
 			trans.logger.Errorf("Data sent to transformer : %v", string(rawJSON))
 			trans.logger.Errorf("Transformer returned : %v", string(respData))
-
-			respData = []byte(fmt.Sprintf("Failed to unmarshal transformer response: %s", string(respData)))
-
-			transformerResponses = nil
-			statusCode = http.StatusBadRequest
+			panic(err)
```

Comment:
PR description and title are quite misleading, right? You're not just adding a missing stat.
How come we are panicking now? Can you add some context in the PR description please?

Reply:
I just added some clean up too .. Will add description for the same